src/components/OpenSeadragonViewer.js

```diff
@@ -118,7 +118,11 @@ export class OpenSeadragonViewer extends Component {
     ) {
       viewer.close();
       const canvasesChanged = !(isEqual(canvasWorld.canvasIds, prevProps.canvasWorld.canvasIds));
-      this.addAllImageSources((canvasesChanged || !viewerConfig));
+      if (canvasesChanged && viewer.preserveViewport) {
+        this.addAllImageSources(false);
+      } else {
+        this.addAllImageSources((canvasesChanged || !viewerConfig));
```

Comment:
I found this negative logic a bit hard to follow (not from what you wrote, but from what existed before!) -- what do you think about adding comments similar to:

```
 if (canvasesChanged && viewer.preserveViewport) {
        // Do not reset the zoom
        this.addAllImageSources(false);
      } else {
       // Reset the zoom if the canvas has changed or if there is no viewerConfig
        this.addAllImageSources((canvasesChanged || !viewerConfig));
      }
...
```
 ?